test/integration/expression-tests/in/basic-array/test.json

```diff
@@ -0,0 +1,34 @@
+{
+  "expression": [
+    "boolean",
+    ["in", ["get", "i"], ["get", "arr"]]
+  ],
+  "inputs": [
+    [{}, {"properties": {"i": null, "arr": [9, 8, 7]}}],
+    [{}, {"properties": {"i": 1, "arr": [9, 8, 7]}}],
+    [{}, {"properties": {"i": 9, "arr": [9, 8, 7]}}],
+    [{}, {"properties": {"i": "foo", "arr": ["baz", "bar", "hello", "foo", "world"]}}],
+    [{}, {"properties": {"i": true, "arr": ["foo", 123, null, 456, false, {}, true]}}],
+    [{}, {"properties": {"i": 1, "arr": null}}]
+  ],
+  "expected": {
+    "compiled": {
+      "result": "success",
+      "isFeatureConstant": false,
+      "isZoomConstant": true,
+      "type": "boolean"
+    },
+    "outputs": [
+      false,
+      false,
+      true,
+      true,
+      true,
+      false
```

Comment:
The difference is in whether or not we're asserting that the value is a string or array. If we do `['string', ['get', 'str']]` and `str` is `null`, then we get a type error. But if we don't assert it's a string or array and just directly return `null` then the check on line 65 of `in.js` returns `false` because there's no value that can be checked